src/posix.mak

```diff
@@ -22,9 +22,15 @@ INSTALL_DIR=../../install
 SYSCONFDIR=/etc
 PGO_DIR=$(abspath pgo)
 
-C=backend
-TK=tk
-ROOT=root
+D = ddmd
+
+C=$D/backend
+TK=$D/tk
+ROOT=$D/root
+
+GENERATED = ../generated
+G = $(GENERATED)/$(OS)$(MODEL)
+$(shell mkdir -p $G)
```

Comment:
> This is interesting. A few explanations follow.
> 
> 2. The structure generated/$(OS)$(MODEL) (as opposed to Phobos' generated/$(OS)/$(MODEL)) was chosen by @RazvanN7 and I after discussing for a while and figuring there's no real need for two separate levels in the dir hierarchy.

I doesn't cost anything either, and consistency across repos seems more important (in particular for scripting, e.g. release scripts). So let's please just stick with `generated/$OS/$BUILD/$MODEL` as anywhere else.

> Even on a given OS, there may be multiple "OS"s - consider generating Windows and Cygwin binaries on Windows. Are there any others? Not sure how important this is, but putting the OS in the dirname helps such situations.

I also allows to more easily rsync code from other machines/VMs.

> Now (this is getting to @wilzbach's point), we have the situation in which changing CFLAGS and DFLAGS affects the generated binaries in many ways, per https://github.com/dlang/dmd/blob/master/src/posix.mak#L160. The "total" solution to that is to canonicalize and place the controlling flags in the dir names, i.e. (after removing ENABLE) in alphabetical order: COVERAGE, DEBUG, LTO, PGO_GENERATE, PGO_USE, PROFILE, PROFILING, RELEASE, UNITTEST.

Those are hardly maintained and rarely used, let's not overshoot and leave it to maintainers.
Separating debug and relase builds is quite helpful though.
An alternative would be to dump all flags to some file and error on mismatch (requiring a clean step). This is the usual configure/Make approach to changing flags.